### DISCUSSION TITLE
How to get reference of an inner object in python?

### DISCUSSION THREAD

**Discussion by sun-rs** [PyO3/pyo3#3020] [Questions] [answered]
```rust
#[pyclass]
struct Parent {
    child: Child,
}

#[pymethods]
impl Parent {
    #[new]
    fn new() -> Self {
        Parent {
            child: Child { value: 0 },
        }
    }

    fn set_age_by_parent(&mut self, age: i32) {
        self.child.set_age(age);
    }
}

#[pyclass]
struct Child {
    value: i32,
}

#[pymethods]
impl Child {
    fn set_age(&mut self, age: i32) {
        self.value = age;
    }
}
```
After wrap these code and generate py module, I could make a new `Parent` in python and call `set_age_by_parent`. But if do `parent.child.set_age(10)`, it seems I will get a copy of `parent.child`. This is not like get a child class from parent class in pure python. Is there any way to do that?

**Comment by davidhewitt** [comment, marked as answer]
You should write

```rust
#[pyclass]
struct Parent {
    child: Py<Child>,
}
```

i.e. `Parent` will store a Python reference to a `Child` object, instead of just the Rust struct.
